EIPS/eip-2794.md

```diff
@@ -0,0 +1,94 @@
+- Virtual trading pairs are established.
+- NDAO will be forged in two ways: realtime forging and staking with forged NDAO
+Realtime forging --- forge NDAOs with ETHs. People who get NDAOs can apply them in various applications.Staking with forged NDAO --- For example, in ETH 2.0 staking, a validator needs to stake 32 ETHs. By using staking with forged NDAO a validator can stake NDAOs instead of ETHs by sending ETHs to the system in exchange for NDAOs as stake. When the validator exits he/she will get back his/her staked NDAOs.
+- The RICP's algorithm runs this way:
+It is initialized with ETH's circulating supply.
+When a trader sends an amount of ETHs to the system, the ETH's circulating supply will be deducted by this amount and the total supply of ETH in the trading pair will be deducted by this amount. We can think of as these ETHs are burned.
+For instance if ETH's total supply is 107,477,373 , a trader sends 100 ETHs to this system in exchange for NDAOs, then the total supply of ETH in the algorithm will be 107,477,273 which will be less than the previous supply by 100, and the ETH's price will rise based on the RICP algorithm.
+- The total supply of ETH in the system and the total supply of NDAO are calculated once on a daily basis.
+- In this system ETH is priced in NDAO.
+
+Note: right after the foring contracts start, NDAO's pegging relationship with USD will be revoked.
+## Backwards Compatibility
+[EIP-2569](https://github.com/DAism2019/EIPs/blob/master/EIPS/eip-2569.md).
+
+[EIP-2569 Pull](https://github.com/ethereum/EIPs/pull/2569).
+
+## Conclusion
+Our proposal presents a solution for mass exit of validators and eliminates two consequences that will cause devastating damages to ETH 2.0.
+- Active validators are tempted to exit and sell staked ETHs due to ETH's price hikes.
+- Active validators will get rewards when they perform an exit. Their rewards will not be associated with ETH's price hikes. Thus validators will not be tempted to speculate on ETH and will not be tempted to perform an exit for this speculation. 
+ETH's price hikes will not affect the real cost (the number of ETHs) for potential validators. ETH's price will not be positively associated with the cost of staking. In the long term a validator's income will increase and more people will likely join staking but the number of ETHs needed for staking for a validator will decrease. As more validators serve in the system the aforementioned extreme scenario will less likely happen.
+This proposal motivates more people to be validators and enhances validators' loyalty(work and behave properly) by rewarding validators with honor tokens. These honor tokens (including their logo images) will be permanently saved on Ethereum. This is the first ever proposal to reward Ethereum's contributors with honor and will greatly incentivize validators.
+
+All in all, this proposal provides a solution for security and safety of ETH 2 Staking.
```

Comment:
This document doesn't fully follow the format of the [Template](https://github.com/ethereum/EIPs/blob/master/eip-template.md).  Can you please update it to match the template, including all necessary sections (like the copyright, the security section, specification, etc.)?

Reply:
I'm working with my partner but we're busy. We just need time to finish it. It's good to tell people in advance that there's a potential risk and also there's a better solution described. I mean at least we have described the solution.